internal/controller/topic/topic.go

```diff
@@ -203,10 +262,87 @@ func (c *external) Update(ctx context.Context, mg resource.Managed) (managed.Ext
 		return managed.ExternalUpdate{}, errors.Wrapf(p.Err, "cannot get topic")
 	}
 
+	tc, err := c.kafkaClient.DescribeTopicConfigs(ctx, meta.GetExternalName(cr))
+	if err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+	rc, err := tc.On(meta.GetExternalName(cr), nil)
+	if err == errMissing(meta.GetExternalName(cr)) {
+		return managed.ExternalUpdate{}, err
+	}
+	if errors.Is(rc.Err, kerr.UnknownTopicOrPartition) {
+		return managed.ExternalUpdate{}, err
+	}
+	if rc.Err != nil {
+		return managed.ExternalUpdate{}, errors.Wrapf(rc.Err, "cannot get topic")
+	}
+
+	provided := cr.Spec.ForProvider.Config
+	if provided != nil {
+		for k, v := range provided {
+			for _, y := range rc.Configs {
+				if k == y.Key {
+					// if provided value is != to set value set the value
+					if v != *y.Value {
+						s := kadm.AlterConfig{
+							Op:    kadm.SetConfig, // Op is the incremental alter operation to perform.
```

Comment:
It may be worth looking into which configs need to be appended or deleted, and whether set works for them. I haven't needed to append or delete before, so I'm not sure which keys are relevant.